src/charm.py

```diff
@@ -189,11 +202,11 @@ def _install_vault_snap(self) -> None:
             vault_snap = snap_cache[VAULT_SNAP_NAME]
             if vault_snap.latest:
                 return
-            self.unit.status = MaintenanceStatus("Installing Vault")
-            vault_snap.ensure(
-                snap.SnapState.Latest, channel=VAULT_SNAP_CHANNEL, revision=VAULT_SNAP_REVISION
-            )
-            vault_snap.hold()
+            with self.temp_maintenance_status("Installing Vault"):
+                vault_snap.ensure(
+                    snap.SnapState.Latest, channel=VAULT_SNAP_CHANNEL, revision=VAULT_SNAP_REVISION
+                )
+                vault_snap.hold()
```

Comment:
This looks awesome, but does it really work? I mean, in juju status, will we see the maintenance status while the snap is installing and then a change to active, all during the same hook?

Reply:
I got surprised by this as well with a different change recently, but yes:

"Changes to status take effect immediately, unlike other Juju operations such as modifying relation data or secrets, which only take effect after a successful event."

Ref: https://ops.readthedocs.io/en/latest/

Reply:
Yep, tried and tested (manually). I set some different statuses with `sleep` and it works as expected.